### ISSUE TITLE
XIP-4: XDAG payment schema proposal

### ISSUE THREAD

**Issue by yangyin5127** [XDagger/XIPs#4]
should add a xip for making xdag payments, witch enable users to easily make payments by simply clicking links on webpages or  QR Codes.

```
 xdagurn     = "xdag:" xdagaddress [ "?" xdagparams ]
 xdagaddress = *base64
 xdagparams  = xdagparam [ "&" xdagparams ]
 xdagparam   = [ amountparam / labelparam / messageparam / otherparam / reqparam ]
 amountparam    = "amount=" *digit [ "." *digit ]
 abelparam     = "label=" *qchar
 messageparam   = "message=" *qchar
 otherparam     = qchar *qchar [ "=" *qchar ]
 reqparam       = "req-" qchar *qchar [ "=" *qchar ]
```

### Examples

Just the address:
```
  xdag:4f1Sp/UD55JX5+kQCevUCpyenaPwqmpC
```
Address with name:
```
 xdag:4f1Sp/UD55JX5+kQCevUCpyenaPwqmpC?label=Solar
```
Request 10.5 XDAG to "Solar":
```
 xdag:4f1Sp/UD55JX5+kQCevUCpyenaPwqmpC?amount=10.5&label=Solar
```
Request 50 XDAG with message:

```
 xdag:4f1Sp/UD55JX5+kQCevUCpyenaPwqmpC?amount=50&label=Solar&message=Donation%20for%20project%20xyz
```

**Comment by xrdavies** [comment]
@feiin 

I suggest to just keep amount, label and message currently. For the rest, we can handle them later.
With new release of xdag, we implement remark field for transactions. This field is used for message.

**Comment by xrdavies** [comment]
Implemented in #5
